neo/Consensus/ConsensusService.cs

```diff
@@ -85,8 +85,11 @@ private bool CheckPolicy(Transaction tx)
             return true;
         }
 
-        private void CheckSignatures()
+        private void OnCommitAgreement(ConsensusPayload payload, CommitAgreement message)
         {
+            if (context.State.HasFlag(ConsensusState.BlockSent)) return;
+            if (!context.TryToCommit(payload, message)) return;
+
             if (context.Signatures.Count(p => p != null) >= context.M && context.TransactionHashes.All(p => context.Transactions.ContainsKey(p)))
```

Comment:
Is this double check necessary? Because ` CheckSignatures()` already checked it and called OnCommitAgreement.